source/Halibut/HalibutRuntime.cs

```diff
@@ -192,8 +197,8 @@ public TService CreateClient<TService>(ServiceEndPoint endpoint)
         
         public TService CreateClient<TService>(ServiceEndPoint endpoint, CancellationToken cancellationToken)
         {
-            messageSerializer.AddToMessageContract(typeof(TService));
-            
+            typeRegistry.AddToMessageContract(typeof(TService));
```

Comment:
The only reason the typeRegistry exists as a member variable here is so that we can ensure the service exists here.

The second 'non-obsolete' constructor expects a type registry, and this class expects it to be used in the messageSerializer.

That doesn't quite feel right to me, as someone could plug in the wrong type registry.